.github/workflows/build.yml

```diff
@@ -174,11 +174,17 @@ jobs:
           tutor plugins enable picasso
 
       - name: Setup SSH agent for private repositories cloning
+        env:
+          SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
+        if: env.SSH_PRIVATE_KEY != ''
         uses: webfactory/ssh-agent@v0.9.0
         with:
             ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
 
       - name: Add GitHub to known hosts
+        env:
```

Comment:
Are we confident the workflow only uses SSH when a key is provided? I’m a bit concerned that gating this step with the same condition could bite us later if something ends up using SSH without a key, though that seems very unlikely

Also, we don’t need to pass the secret into `env` just to check it

```yaml
- name: Add GitHub to known hosts
  if: ${{ secrets.SSH_PRIVATE_KEY }}
  run: |
...
```

Reply:
Yes @magajh I have tested this approach in a production environment, and I'm confident it's working well. The only time we use SSH is to download private packages, which means we need to set up an SSH private key.